[IMP] inventory: overhaul 3rd party shippers

Felicious opened the pull request:
## Summary of changes
1. Overhaul 8 year old 3rd party shipping carrier doc to 16.0
2. Addresses @moch-odoo's [suggestions](https://www.odoo.com/web#id=3576549&cids=3&menu_id=4720&action=333&active_id=3835&model=project.task&view_type=form) to update shipping connector installation instructions
3. Add a table of shipping methods and region availability in Odoo  

## FW port
Yes, all the way to master (:

[Task](https://www.odoo.com/web#id=3576549&cids=3&menu_id=4720&action=333&active_id=3835&model=project.task&view_type=form)

robodoo commented:
[Pull request status dashboard](https://mergebot.odoo.com/odoo/documentation/pull/8634).

brse-odoo commented on content/applications/inventory_and_mrp/inventory/shipping_receiving/setup_configuration/third_party_shipper.rst:
```diff
@@ -1,161 +1,208 @@
-The configuration of the shipper is split into two columns:
+On the shipping method page, configure the :doc:`delivery method details
+<../setup_configuration/delivery_method>`, such as :guilabel:`Integration Level`,
+:guilabel:`Delivery Product`, and :guilabel:`Invoicing Policy`.
 
--  The first one is linked to **your account** (developer key, password,...).
-   For more information, please refer to the provider website.
+Then, in the :guilabel:`Configuration` tab, input the API credentials and password with the shipping
+carrier. Then, go to the :guilabel:`Destination Availability` tab and configure the service coverage
+of the shipping carrier.
```
For these instructions here, do you think some additional instruction on what these are, how you find them, and what they affect would be helpful? I know it's different per carrier, but maybe a .. note:: or a .. tip:: saying "read the doc for individual carriers to find this stuff" would be nice

Felicious replied:
```suggestion
of the shipping carrier.  For specific details about each
shipping method, refer to the following documents:

.. seealso::
   - :doc:`DHL credentials <dhl_credentials>`
   - :doc:`Sendcloud credentials <sendcloud_shipping>`
   - :doc:`UPS credentials <ups_credentials>
```

brse-odoo commented on content/applications/inventory_and_mrp/inventory/shipping_receiving/setup_configuration/third_party_shipper.rst:
```diff
@@ -1,161 +1,208 @@
-domestic providers. For example, USPS US only delivers from the United
-States to the United States.
+.. warning::
+   Set the delivery method to :guilabel:`Production` charges customers for shipping. Verify all
```
```suggestion
   Setting the delivery method to :guilabel:`Production` charges customers for shipping. Verify all
```

brse-odoo commented on content/applications/inventory_and_mrp/inventory/shipping_receiving/setup_configuration/third_party_shipper.rst:
```diff
@@ -1,161 +1,208 @@
-States to the United States.
+.. warning::
+   Set the delivery method to :guilabel:`Production` charges customers for shipping. Verify all
+   configurations before launching the delivery method to :guilabel:`Production`.
```
```suggestion
   configurations are correct before launching the delivery method to :guilabel:`Production`.
```

brse-odoo commented on content/applications/inventory_and_mrp/inventory/shipping_receiving/setup_configuration/third_party_shipper.rst:
```diff
@@ -1,161 +1,208 @@
 
-The price is computed when you save the sale order or when you click on
-**Set price**.
+Then, switch to the :guilabel:`Inventory` tab, define the :guilabel:`Weight` of the product, in
```
```suggestion
Then, switch to the :guilabel:`Inventory` tab, and define the :guilabel:`Weight` of the product, in
```

brse-odoo commented on content/applications/inventory_and_mrp/inventory/shipping_receiving/setup_configuration/third_party_shipper.rst:
```diff
@@ -1,161 +1,208 @@
-The price is computed when you save the sale order or when you click on
-**Set price**.
+Then, switch to the :guilabel:`Inventory` tab, define the :guilabel:`Weight` of the product, in
+kilograms.
```
Why kilograms? Does it have to be? Can different products have different UoMs?

Felicious replied:
You're right! I confirmed with an expert, and this field has become really flexible since 8 years ago, when the line was written (:

brse-odoo commented on content/applications/inventory_and_mrp/inventory/shipping_receiving/setup_configuration/third_party_shipper.rst:
```diff
@@ -1,161 +1,208 @@
-information from it. You can change the carrier if you want to.
+After configuring the third-party carrier's :ref:`delivery method
+<inventory/shipping_receiving/configure-delivery-method>` in Odoo, navigate to a quotation by going
+to :menuselection:`Sales app --> Orders --> Quotations`.
```
Or, we can create a new quotation also right?

brse-odoo commented on content/applications/inventory_and_mrp/inventory/shipping_receiving/setup_configuration/third_party_shipper.rst:
```diff
@@ -1,161 +1,208 @@
 
-The weight is computed based on the products weights. The shipping cost
-will be computed once the transfer is validated.
+In the pop-up window, select the intended carrier from the :guilabel:`Shipping Method` drop-down
```
```suggestion
In the resulting pop-up window, select the intended carrier from the :guilabel:`Shipping Method` drop-down
```

brse-odoo commented on content/applications/inventory_and_mrp/inventory/shipping_receiving/setup_configuration/third_party_shipper.rst:
```diff
@@ -1,161 +1,208 @@
+delivery order.
+
+In the :guilabel:`Additional info` tab, the third-party shipping carrier is listed in the
+:guilabel:`Carrier` field. When the delivery method set to :ref:`production mode
```
```suggestion
:guilabel:`Carrier` field. When the delivery method is set to :ref:`production mode
```

brse-odoo reviewed (approved):
@Felicious I've finished my review of this PR, great job! I'm approving with a few comments for you to accept/reject as you see fit. Let me know if you have any questions, thanks!

Felicious commented:
Hi @odoo/us-doc-review ! This doc is ready for review :smile:

ksc-odoo commented on content/applications/inventory_and_mrp/inventory/shipping_receiving/setup_configuration/third_party_shipper.rst:
```diff
@@ -1,161 +1,240 @@
+--------------
+
+For users making shipments without installing the *Sales* app, assign the shipping carrier to the
+delivery order, by first going to the :menuselection:`Inventory` app. Select the :guilabel:`Delivery
```
```suggestion
delivery order, by first going to the :menuselection:`Inventory` app. Then, from the :guilabel:`Inventory Overview` dashboard, select the :guilabel:`Delivery
```

ksc-odoo commented on content/applications/inventory_and_mrp/inventory/shipping_receiving/setup_configuration/third_party_shipper.rst:
```diff
@@ -1,161 +1,240 @@
+
+For users making shipments without installing the *Sales* app, assign the shipping carrier to the
+delivery order, by first going to the :menuselection:`Inventory` app. Select the :guilabel:`Delivery
+Orders` operation type, and choose the desired delivery order.
```
do you think it would be helpful to mention that the user should choose a DO that isn't already marked as 'Done' -- or do you think that is redundant info?

Felicious replied:
```suggestion
Orders` operation type, and choose the desired delivery order that is not already marked as :guilabel:`Done` or :guilabel:`Cancelled`.
```

ksc-odoo commented on content/applications/inventory_and_mrp/inventory/shipping_receiving/setup_configuration/third_party_shipper.rst:
```diff
@@ -1,161 +1,240 @@
+delivery order, by first going to the :menuselection:`Inventory` app. Select the :guilabel:`Delivery
+Orders` operation type, and choose the desired delivery order.
+
+In the :guilabel:`Additional info` tab, set :guilabel:`Carrier` field to the third-party shipping
```
```suggestion
In the :guilabel:`Additional info` tab, set :guilabel:`Carrier` field to the desired third-party shipping
```

ksc-odoo reviewed (changes requested):
Hey @Felicious -- just finished my Final Review. Great work on this so far. I think we're *just* about at the finish line. Once you review my feedback, and make the necessary adjustments, feel free to tag me again for another quick look. Thanks! 👍

Felicious commented on content/applications/inventory_and_mrp/inventory/shipping_receiving/setup_configuration/third_party_shipper.rst:
```diff
@@ -1,161 +1,240 @@
 
-Overview
-========
+Link third-party shipping carriers to Odoo databases to verify carriers' delivery to specific
+addresses, :doc:`automatically calculate shipping costs <delivery_method>`, and :doc:`generate
+shipping labels <../advanced_operations_shipping/labels>`.
```
Very nice rewrite, thank you!

Felicious commented on content/applications/inventory_and_mrp/inventory/shipping_receiving/setup_configuration/third_party_shipper.rst:
```diff
@@ -1,161 +1,240 @@
+- :guilabel:`Shipping Method` (*Required field*): the name of the delivery method (e.g. `FedEx US`,
+  `FedEx EU`, etc.).
+- :guilabel:`Provider` (*Required field*): choose the delivery service, like Fedex.
+- :guilabel:`Integration Level`: choose :guilabel:`Get Rate` to simply get an :ref:`estimated
```
These are good points! Taking note for myself here: upon selecting a Provider, the following fields appear:

Integration Level
Invoicing Policy
Insurance Percentage
Configuration tab

Felicious commented on content/applications/inventory_and_mrp/inventory/shipping_receiving/setup_configuration/third_party_shipper.rst:
```diff
@@ -1,161 +1,240 @@
 
--  Uncheck **Test Mode** when you are done with the testings.
+With the delivery method details configured, click the :guilabel:`Test Environment` smart button to
+set it to production environment.
```
```suggestion
set it to :guilabel:`Production Environment`.
```

Felicious commented on content/applications/inventory_and_mrp/inventory/shipping_receiving/setup_configuration/third_party_shipper.rst:
```diff
@@ -1,161 +1,240 @@
+   :alt: Show the "Add shipping" button at the bottom of a quotation.
+
+In the resulting pop-up window, select the intended carrier from the :guilabel:`Shipping Method`
+drop-down menu. The :guilabel:`Cost` field is automatically filled based on the :ref:`product weight
+<inventory/shipping_receiving/configure-weight>` and distance between the warehouse's :ref:`source
```
```suggestion
drop-down menu. The :guilabel:`Cost` field is automatically filled based on:
- the amount specified in the :guilabel:`Total Order Weight` field (if it is not provided, the sum of :ref:`product weights <inventory/shipping_receiving/configure-weight>` in the order is used) 
- the distance between the warehouse's :ref:`source
```

Felicious commented on content/applications/inventory_and_mrp/inventory/shipping_receiving/setup_configuration/third_party_shipper.rst:
```diff
@@ -1,161 +1,240 @@
-Click on **Validate** to receive the tracking number. The **Carrier Tracking
-ref** field will automatically be filled. Click on the **Tracking** button
-to check your delivery on the provider website.
+Click :guilabel:`Get Rate` in the :guilabel:`Add a shipping method` pop-up window to get the
```
```suggestion
After selecting a third party provider in the :guilabel:`Shipping Method` field, click :guilabel:`Get Rate` in the :guilabel:`Add a shipping method` pop-up window to get the
```

ksc-odoo commented on content/applications/inventory_and_mrp/inventory/shipping_receiving/setup_configuration/third_party_shipper.rst:
```diff
@@ -1,161 +1,268 @@
 
-Install the shipper company connector module
---------------------------------------------
+To ensure proper setup of the third-party shipping carrier with Odoo, follow these steps:
```
```suggestion
To ensure proper setup of a third-party shipping carrier with Odoo, follow these steps:
```

ksc-odoo commented on content/applications/inventory_and_mrp/inventory/shipping_receiving/setup_configuration/third_party_shipper.rst:
```diff
@@ -1,161 +1,268 @@
-States to the United States.
+.. warning::
+   Setting the delivery method to :guilabel:`Production` creates **real** shipping labels, and users
+   are at risk for being charged through their carrier account (e.g. UPS, FedEx, etc.), **before**
```
```suggestion
   are at risk of being charged through their carrier account (e.g. UPS, FedEx, etc.) **before**
```

ksc-odoo commented on content/applications/inventory_and_mrp/inventory/shipping_receiving/setup_configuration/third_party_shipper.rst:
```diff
@@ -1,161 +1,268 @@
+   :align: center
+   :alt: Highlight the "Company" field.
+
+Verify the :guilabel:`Address` and :guilabel:`Phone` number are correct, as they are required for
```
```suggestion
Verify that the :guilabel:`Address` and :guilabel:`Phone` number are correct, as they are required for
```

ksc-odoo commented on content/applications/inventory_and_mrp/inventory/shipping_receiving/setup_configuration/third_party_shipper.rst:
```diff
@@ -1,161 +1,268 @@
 
-The price is computed when you save the sale order or when you click on
-**Set price**.
+To assign a third party shipping carrier, and get an estimated cost of shipping, begin by going to
```
```suggestion
To assign a third-party shipping carrier, and get an estimated cost of shipping, begin by going to
```

ksc-odoo commented on content/applications/inventory_and_mrp/inventory/shipping_receiving/setup_configuration/third_party_shipper.rst:
```diff
@@ -1,161 +1,268 @@
 
-If you create a delivery transfer from the inventory module, you can add
-the third party shipper in the additional info tab.
+After selecting a third party provider in the :guilabel:`Shipping Method` field, click
```
```suggestion
After selecting a third-party provider in the :guilabel:`Shipping Method` field, click
```

ksc-odoo commented on content/applications/inventory_and_mrp/inventory/shipping_receiving/setup_configuration/third_party_shipper.rst:
```diff
@@ -1,161 +1,268 @@
 
-.. image:: third_party_shipper/third_party_shipper04.png
-   :align: center
+In the :guilabel:`Additional info` tab, set :guilabel:`Carrier` field to the desired third-party
```
```suggestion
In the :guilabel:`Additional info` tab, set the :guilabel:`Carrier` field to the desired third-party
```

ksc-odoo reviewed (approved):
Okie dokie, @Felicious -- after you address the few comments I've left, this should be ready to be tagged for Tech Review. Approving now! 👍

Felicious commented:
Hello @samueljlieber ! This PR is ready for your technical review :smile: Thank you, as always!

Felicious commented:
Hi @samueljlieber ! Thank you so much for reminding me about the media file convention -- totally forgot about to reference it. Appreciate your attention to detail, as always! :grin: 

I've addressed all your comments, and the PR's ready for a second review

samueljlieber reviewed (approved):
Nice job @Felicious, your changes look good to me! Nice job :)
.....
@robodoo r+
